cpp/src/arrow/io/hdfs.h

```diff
@@ -161,6 +161,8 @@ class ARROW_EXPORT HadoopFileSystem : public FileSystem {
   Status OpenWriteable(const std::string& path, bool append,
                        std::shared_ptr<HdfsOutputStream>* file);
 
+  Status GetFileBlockLocations(const std::string& path , int64_t offset, int64_t size , std::vector<std::vector<std::string>> * block_locations);
```

Comment:
Can you add a Doxygen-style comment for this new API?

The problem with the API is that there is no context about how to interpret the results. We need to know for each values in the returned vector:

* the start position in the file
* the length of the block
* the host where the block is located

Since we can find out the file size and block size from the file metadata, we can make multiple queries to GetHosts to determine this information for each HDFS block. With the current implementation, if `size` spans more than one block we will not have this information. It probably makes sense to expose a struct `HdfsBlockInfo` which has this data

Reply:
Okay. I will add the doxygen-style comment for the API and Python binding for this.

Yes, It make sense to expose a HdfsBlockInfo. Will work toward it now.

Reply:
I have updates the pull request with changes required to expose a HdfsBlockInfo
TODO:
1) Doxygen Style Comment